src/client/containers/DeleteBoardModal/DeleteBoardModal.container.js

```diff
@@ -1,21 +1,29 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import './DeleteBoardModal.style.css';
 import CloseButton from '../../components/CloseButton/CloseButton.component';
 import PageTitle from '../../components/PageTitle/PageTitle.component';
-import GenericButton from '../../components/GenericButton/GenericButton.component';
+import GenericButton from '../../statcomponents/GenericButton/GenericButton.component';
+import { ApiError } from '../../ErrorBoundary';
+
+const DeleteBoardModal = ({ boardInfo, setModalVisibility, userId }) => {
+  // eslint-disable-next-line no-unused-vars
+  const [error, setError] = useState(null);
+  const boardId = boardInfo.id;
 
-const DeleteBoardModal = ({ boardInfo, setModalVisibility }) => {
-  const userId = 2;
-  const boardId = 1;
   const deleteBoard = async () => {
     const API_URL = `/api/users/${userId}/boards/${boardId}`;
     try {
-      await fetch(API_URL, { method: 'DELETE' });
+      const response = await fetch(API_URL, { method: 'DELETE' });
       // eslint-disable-next-line no-alert
-      alert('Data deleted');
-    } catch (error) {
-      throw new Error(error);
+      if (!response.ok) {
+        throw new ApiError(response.statusText, response.status);
+      }
+    } catch (err) {
+      setError(() => {
+        // eslint-disable-next-line new-cap
+        throw new ApiError(err.message, err.statusCode);
```

Comment:
Great 👍
We really need a global fetch method to do this for us.
Try and provoke an error and check if its working. Merge when done